### DISCUSSION TITLE
Open dialog for "existing image" menu items should default to vmpath

### DISCUSSION THREAD

**Discussion by charabaruk** [86Box/86Box#4005] [Ideas]
When I launch 86Box via 86BoxManager and then go to attach an existing image to one of the VM's configured drives, the first time the Open dialog appears it is always in the directory where I have 86BoxManager installed. It would make more sense for the dialog to default to the VM's own directory as set by the --vmpath command line option or equivalent until I change the directory in the dialog to somewhere else.

**Comment by charabaruk** [comment]
I just found that there's an option "Select media images from program working directory" in the Preferences dialog. I guess this ask really is to make that option enabled by default.

**Comment by cold-brewed** [comment]
Note that this behavior varies by OS because of qt, which is used to provide the file selection dialog.

Qt by default uses the working directory where the program was launched unless otherwise specified. This of course varies by OS and launch method. The option you found just specifies the directory to Qt.

This is even more apparent on macOS which will sometimes open the file dialog with a default to root if launched through Finder.

Most likely if you were to launch it directly (as in, not through the manager) you'd see that the image selection dialog would be from wherever you launched it.

All that said, maybe @daviunic could confirm for us whether or not 86BoxManager changes the working directory before launching.

**Reply by charabaruk** [reply]
I just checked in Process Explorer, and 86Box.exe was started in the directory where 86BoxManager was started.

QFileDialog _does_ have a method to explicitly set the directory shown when the dialog opens, so it should be possible to make this work. If QFileDialog::setDirectory() is passed the value of vmpath, and the dialog _doesn't_ open to that directory, that would be a bug on Qt.

**Reply by cold-brewed** [reply]
Yes, the option in preferences does exactly that: Passes the value of `usr_path` to `QFileDialog` to have it default to that directory.
